tests/test_java.py

```diff
@@ -718,3 +720,284 @@ def test_java_frames_include_no_semicolons(
+                {
+                    "name": "InitialHeapSize",
+                    "type": "size_t",
+                    "value": "264241152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MarkStackSize",
+                    "type": "size_t",
+                    "value": "4194304",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxHeapSize",
+                    "type": "size_t",
+                    "value": "4196401152",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxNewSize",
+                    "type": "size_t",
+                    "value": "2517630976",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MinHeapDeltaBytes",
+                    "type": "size_t",
+                    "value": "1048576",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SegmentedCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "UseCompressedClassPointers",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {
+                    "name": "UseCompressedOops",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {"name": "UseG1GC", "type": "bool", "value": "true", "origin": "ergonomic", "kind": ["product"]},
+            ],
+        ),
+    ],
+)
+def test_get_default_jvm_flags(
+    tmp_path: Path,
+    application_pid: int,
+    assert_collapsed: AssertInCollapsed,
+    expected_flags: List[Dict[str, Union[str, List[str]]]],
+) -> None:
+    with make_java_profiler(
+        frequency=99,
+        storage_dir=str(tmp_path),
+        java_async_profiler_mode="cpu",
+    ) as profiler:
+        assert_jvm_flags_equal(
+            profiler._metadata.get_jvm_flags_serialized(psutil.Process(application_pid)), expected_flags
+        )
+
+
+@pytest.mark.parametrize(
+    "java_cli_flags,java_env_flags",
+    [
+        (
+            "-XX:SelfDestructTimer=5",
+            "-XX:+PrintCodeCache",
+        ),
+    ],
+)
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize(
+    "application_image_tag,expected_flags",
```

Comment:
If the `expected_flags` is constant across different images, don't parametrize it alongisde the image. Parametrize it once OR keep it a const in the test.

Reply:
I see there are slight differences between JDK 8 and JDK 11, so up to you if u want to keep it here or reduce the boilerplate somehow

Reply:
totally different, since JDK11 flags contains "origin" whilst JDK8 only specify "`default"` or `"non-default"`